File: tokit/tokit/tool/tick_tool.cpp
///<------------------------------------------------------------------------------
//< @file:   ticktool.cpp
//< @author: 洪坤安
//< @date:   2014年9月20日 11:46:36
//< @brief:  
//< Copyright (c) 2014 Tokit. All rights reserved.
///<------------------------------------------------------------------------------

#include "tick_tool.h"

#include <windows.h>

namespace ticktool
{
    // 获取CPU每秒的滴答次数
    Tick GetTickFrequency()
    {
        static LARGE_INTEGER static_perfFreq = {0};
        if(0 == static_perfFreq.QuadPart){
            QueryPerformanceFrequency(&static_perfFreq); 
        }

        return static_perfFreq.QuadPart;
    }

    Tick get_tick()
    {
        LARGE_INTEGER tick;

        QueryPerformanceCounter(&tick);
        return tick.QuadPart;
    }

    uint32_t tick_diff(Tick old_tick)
    {
        Tick tick_now = get_tick();

        uint32_t ms = (uint32_t)((tick_now - old_tick) * 1000 / GetTickFrequency());
        return ms;
    }
}

// 返回秒
double tick_t::end_tick()
{
    Tick tick_now = ticktool::get_tick();
    Tick diff = tick_now - m_tick;

    // uint32_t ms = (double)((tick_now - old_tick) * 1000 / GetTickFrequency());

    double passed_sec = (double)diff / ticktool::GetTickFrequency();
    return passed_sec;
}

File: tokit_go_src/tokit/tool/tick_tool.go
///<------------------------------------------------------------------------------
//< @文件名：tick_tool.cpp
//< @作　者：洪坤安
//< @日　期: 2014年11月12日 14:55:42
//< @摘　要:
//< Copyright (c) 2014 Tokit. All rights reserved.
///<------------------------------------------------------------------------------

// tick_tool
package tool

import (
	"time"
)

type Clock struct
{
    m_tick time.Time
};

func (clock *Clock) Start_tick(){
	clock.m_tick = time.Now()
}

// 返回秒
func (clock *Clock) End_tick() float64 {
	now_tick := time.Now()
	var dur_time time.Duration = now_tick.Sub(clock.m_tick)

	var elapsed_sec float64 = dur_time.Seconds()
	return elapsed_sec
}